src/main/java/io/gravitee/policy/groovy/model/ScriptableExecutionContext.java

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright © 2015 The Gravitee team (http://gravitee.io)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.gravitee.policy.groovy.model;
+
+import io.gravitee.el.TemplateEngine;
+import io.gravitee.gateway.reactive.api.context.GenericExecutionContext;
+import io.gravitee.gateway.reactive.api.context.GenericRequest;
+import io.gravitee.gateway.reactive.api.context.GenericResponse;
+import io.gravitee.reporter.api.v4.metric.Metrics;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import lombok.Value;
+
+/**
+ * @author Antoine CORDIER (antoine.cordier at graviteesource.com)
+ * @author GraviteeSource Team
+ */
+@Value
+public class ScriptableExecutionContext implements GenericExecutionContext {
```

Comment:
I think it should be exclusively read-only and not allow any direct modifications.

Reply:
I had a closer look and I think, at least for attributes modifier, we should keep them as it look they are used a lot. Our integration tests for v3 even make use of them.